Mathlib/Analysis/NormedSpace/WeakDual.lean

```diff
@@ -0,0 +1,258 @@
+functions `E → 𝕜` with the topology of pointwise convergence.
+
+The polar set `polar 𝕜 s` of a subset `s` of `E` is originally defined as a subset of the dual
+`dual 𝕜 E`. We care about properties of these w.r.t. weak-* topology, and for this purpose give
+the definition `weak_dual.polar 𝕜 s` for the "same" subset viewed as a subset of `weak_dual 𝕜 E`
+(a type synonym of the dual but with a different topology instance).
+
+## References
+
+* https://en.wikipedia.org/wiki/Weak_topology#Weak-*_topology
+* https://en.wikipedia.org/wiki/Banach%E2%80%93Alaoglu_theorem
+
+## Tags
+
+weak-star, weak dual
+
+-/
+
+
+noncomputable section
+
+open Filter Function Metric Set
+
+open Topology Filter
+
+/-!
+### Weak star topology on duals of normed spaces
+
+In this section, we prove properties about the weak-* topology on duals of normed spaces.
+We prove in particular that the canonical mapping `dual 𝕜 E → weak_dual 𝕜 E` is continuous,
+i.e., that the weak-* topology is coarser (not necessarily strictly) than the topology given
+by the dual-norm (i.e. the operator-norm).
+-/
+
+
+variable {𝕜 : Type _} [NontriviallyNormedField 𝕜]
+
+variable {E : Type _} [SeminormedAddCommGroup E] [NormedSpace 𝕜 E]
+
+namespace NormedSpace
+
+namespace Dual
+
+/-- For normed spaces `E`, there is a canonical map `dual 𝕜 E → weak_dual 𝕜 E` (the "identity"
+mapping). It is a linear equivalence. -/
+def toWeakDual : Dual 𝕜 E ≃ₗ[𝕜] WeakDual 𝕜 E :=
+  LinearEquiv.refl 𝕜 (E →L[𝕜] 𝕜)
+#align normed_space.dual.to_weak_dual NormedSpace.Dual.toWeakDual
+
+@[simp]
+theorem coe_toWeakDual (x' : Dual 𝕜 E) : toWeakDual x' = x' :=
+  rfl
+#align normed_space.dual.coe_to_weak_dual NormedSpace.Dual.coe_toWeakDual
+
+@[simp]
+theorem toWeakDual_eq_iff (x' y' : Dual 𝕜 E) : toWeakDual x' = toWeakDual y' ↔ x' = y' :=
+  Function.Injective.eq_iff <| LinearEquiv.injective toWeakDual
+#align normed_space.dual.to_weak_dual_eq_iff NormedSpace.Dual.toWeakDual_eq_iff
+
+theorem toWeakDual_continuous : Continuous fun x' : Dual 𝕜 E => toWeakDual x' :=
+  WeakBilin.continuous_of_continuous_eval _ fun z => (inclusionInDoubleDual 𝕜 E z).continuous
+#align normed_space.dual.to_weak_dual_continuous NormedSpace.Dual.toWeakDual_continuous
+
+/-- For a normed space `E`, according to `to_weak_dual_continuous` the "identity mapping"
+`dual 𝕜 E → weak_dual 𝕜 E` is continuous. This definition implements it as a continuous linear
+map. -/
+def continuousLinearMapToWeakDual : Dual 𝕜 E →L[𝕜] WeakDual 𝕜 E :=
+  { toWeakDual with cont := toWeakDual_continuous }
+#align normed_space.dual.continuous_linear_map_to_weak_dual NormedSpace.Dual.continuousLinearMapToWeakDual
+
+/-- The weak-star topology is coarser than the dual-norm topology. -/
+theorem dual_norm_topology_le_weak_dual_topology :
+    (by infer_instance : TopologicalSpace (Dual 𝕜 E)) ≤
+      (by infer_instance : TopologicalSpace (WeakDual 𝕜 E)) := by
+  convert toWeakDual_continuous.le_induced
```

Comment:
I will definitely not be able to fix this as I don't even understand how there are tactics in the statement of the theorem. Any suggestions on how to deal with this?

Reply:
Shouldn't the RHS just be `WeakDual.instTopologicalSpace 𝕜 E`? For the RHS maybe just check what `infer_instance` outputs

Reply:
Here's the goal window with `pp` false:

```lean 
  𝕜 : Type u_1
  inst✝² : NontriviallyNormedField 𝕜
  E : Type u_2
  inst✝¹ : SeminormedAddCommGroup E
  inst✝ : NormedSpace 𝕜 E
  ⊢ ∀ {α β : Type (max ?u.65216 ?u.65217)} {t : TopologicalSpace α} {t' : TopologicalSpace β} {f : α → β},
    Continuous f → t ≤ TopologicalSpace.induced f t'
```